internal/ethapi/api.go

```diff
@@ -1135,23 +1138,9 @@ func (s *PublicTransactionPoolAPI) SendRawTransaction(ctx context.Context, encod
 		return "", err
 	}
 
-	if err := s.b.SendTx(ctx, tx); err != nil {
-		return "", err
-	}
-
-	signer := types.MakeSigner(s.b.ChainConfig(), s.b.CurrentBlock().Number())
-	if tx.To() == nil {
-		from, err := types.Sender(signer, tx)
-		if err != nil {
-			return "", err
-		}
-		addr := crypto.CreateAddress(from, tx.Nonce())
-		log.Info("Submitted contract creation", "fullhash", tx.Hash().Hex(), "contract", addr.Hex())
-	} else {
-		log.Info("Submitted transaction", "fullhash", tx.Hash().Hex(), "recipient", tx.To())
-	}
+	txHash, err := submitTransaction(ctx, s.b, tx)
+	return txHash.Hex(), err
```

Comment:
This would fail if `txHash` is nil, which is a common pattern in case `err` is non-nil. And I think it changes the functionality, now it would return empty hash (zeroes) instead of empty string. Instead: check if err is non-nil, and if so return `"", err`.

Reply:
This is fine and doesn't need change because

* `common.Hash` cannot be nil.
* The RPC server ignores the result and returns the error if it is non-nil.

Reply:
`SendRawTransaction` should return `(common.Hash, error)` and this can be reduced to `return submitTransaction(....)`.